pydantic/main.py

```diff
@@ -229,7 +240,7 @@ def __new__(mcs, name, bases, namespace, **kwargs):  # noqa C901
         new_namespace = {
             '__config__': config,
             '__fields__': fields,
-            '__field_defaults__': {n: f.default for n, f in fields.items() if not f.required},
+            '__field_defaults__': fields_defaults,
```

Comment:
Has `__field_defaults__` changed at all?

Reply:
Nope, just saw that there was unnecessary iteration through fields and replaced it.